### ISSUE TITLE
Missing Parameters & Optimization

### ISSUE THREAD

**Issue by SundousHussein** [UCD-BDLab/BioNeuralNet#24]
There are multiple missing hyperparameters for GNNs and the entire optimization process is not there. 
Missing Hyperparameters: 
- epoch: Number of Epochs to Train the Model.
- lr: Learning Rate. 
- weight_decay: Weight Decay. 
** Potentially other parameters.
https://github.com/UCD-BDLab/BioNeuralNet/blob/571e05c39155b10596353329ffad84106f64d272/bioneuralnet/network_embedding/gnn_embedding.py#L24C8-L32C42
